stellar-auth/pkg/auth/authenticator.go

```diff
@@ -252,29 +259,30 @@ func (a *defaultAuthenticator) DeactivateUser(ctx context.Context, userID string
 }
 
 func (a *defaultAuthenticator) ForgotPassword(ctx context.Context, sqlExec db.SQLExecuter, email string) (string, error) {
+	email = strings.TrimSpace(strings.ToLower(email))
 	if email == "" {
-		return "", fmt.Errorf("error generating user reset password token: email cannot be empty")
+		return "", fmt.Errorf("generating user reset password token: email cannot be empty")
 	}
 
 	resetToken, err := utils.StringWithCharset(resetTokenLength, utils.DefaultCharset)
 	if err != nil {
-		return "", fmt.Errorf("error generating random reset token in forgot password: %w", err)
+		return "", fmt.Errorf("generating random reset token in forgot password: %w", err)
 	}
 
 	checkValidTokenQuery := `
 		SELECT EXISTS (
 			SELECT 1
 			FROM auth_user_password_reset ar
 			INNER JOIN auth_users au ON ar.auth_user_id = au.id
-			WHERE au.email = $1
+			WHERE LOWER(au.email) = LOWER($1)
```

Comment:
in theory, this wouldn't be needed right? Considering your migration is sanitizing the database values and this handler is sanitizing the incoming email value.

Reply:
Yeah I debated this one too because it shouldn't be needed. It felt too defensive to add this check but I thought it wouldn't hurt given that this table isn't expected to be too large. You vote for removing it ?

Reply:
Addressed in fa6fd881592bf